### PR TITLE
boards: arm: bl5340_dvk: Add MCP4725 dac to bl5340_dvk_cpuapp dts file

### PR DESCRIPTION
**Pull request by kieran-mackey**
MCP4725 is an I2C dac that was added with PR #33481. This can now be
added to the bl5340_dvk device tree.

Signed-off-by: Kieran Mackey <kieran.mackey@lairdconnect.com>

### CODE REVIEW

File: drivers/dac/dac_mcp4725.c [hunk 1]
```diff
@@ -139,7 +139,7 @@ static const struct dac_driver_api mcp4725_driver_api = {
 	};								\
 									\
 	DEVICE_DT_INST_DEFINE(index, dac_mcp4725_init,			\
-			    device_pm_control_nop,			\
+			    NULL,					\
```
Comment (galak) [inline drivers/dac/dac_mcp4725.c:142]: Can you split this out into its own PR.  This is a bug that should get fixed in 2.6.0

Comment (kieran-mackey) [reply drivers/dac/dac_mcp4725.c]: I've removed this change from this PR

Comment (galak) [reply drivers/dac/dac_mcp4725.c]: Please do submit a new PR with this in it.

Comment (kieran-mackey) [reply drivers/dac/dac_mcp4725.c]: PR created #35265

File: tests/drivers/dac/dac_loopback/src/test_dac.c [hunk 2]
```diff
@@ -133,6 +133,24 @@
...
+#define DAC_RESOLUTION		12
+#define DAC_CHANNEL_ID		0
+
+#define ADC_DEVICE_NAME		DT_LABEL(DT_NODELABEL(adc))
+#define ADC_RESOLUTION		12
+#define ADC_GAIN		ADC_GAIN_1_4
+#define ADC_REFERENCE		ADC_REF_VDD_1_4
+#define ADC_ACQUISITION_TIME	ADC_ACQ_TIME_DEFAULT
+#define ADC_CHANNEL_ID		1
+#define ADC_1ST_CHANNEL_INPUT	NRF_SAADC_INPUT_AIN2
```
Comment (martinjaeger) [inline tests/drivers/dac/dac_loopback/src/test_dac.c:152]: Can't this be `NRF_SAADC_INPUT_AIN1` as above, so that we don't need a completely new board section? It needs an external connection with a wire anyway, correct? Or is this already wired on the PCB?

Comment (kieran-mackey) [reply tests/drivers/dac/dac_loopback/src/test_dac.c]: Just gave it a quick test and it can (it does use an external connection). The fix has been made.

**Review by martinjaeger — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Review by galak — APPROVED** [review]
